src/app/components/new/new.component.ts

```diff
@@ -22,7 +30,35 @@ export class NewComponent {
     }
     return false;
   }
+
+  checkIfCommandDone(commandId: string): void {
+    this.commandService.isCommandDone(commandId)
+      .subscribe(response => {
+        this.isCommandDone = !!response;
+      });
+  }
+
+  checkAngularProject(): void {
+    this.commandService.isAngularProject()
+      .subscribe(response => this.isAngularProject = !!response);
+  }
+
   createNewProject() {
-    this.sendCommand.emit(this.options.createCommandString());
+    const requrst = new CommandRequest(this.options.createCommandString());
+    this.commandService.sendCommand(requrst)
+      .subscribe(commandId => {
```

Comment:
what if the server returns 400?

Reply:
+ should be in its own method

something like:
```typescript
.subscribe(commandId => this.foo(commandId))
```

Reply:
you should keep the interval `timedStatusCheck` as a class member in order to unsubscribe from it in case of error